Polyfills/Console/Source/Console.cpp

```diff
@@ -34,20 +71,137 @@ namespace
         }
     }
 
+    bool EndsWith(std::string_view s, char c) {
+        if (s.size() > 0)
+        {
+            return s.at(s.size() - 1) == c;
+        }
+        return false;
+    }
+
     void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
         std::stringstream ss{};
-        for (size_t i = 0; i < info.Length(); i++)
+        std::string formattedString{};
+        if (info.Length() > 0)
         {
-            if (i > 0)
+            std::string firstArg = info[0].ToString();
+            std::vector<STRING_PART> parts = std::vector<STRING_PART>();
+
+            // Split the first string into parts limited by substitution characters
+            size_t start = 0;
+            bool matching = false;
+            std::size_t j;
+            for (j = 0; j < firstArg.size(); j++)
+            {
+                const char currChar = firstArg.at(j);
+                if (!matching && currChar == '%')
+                {
+                    matching = true;
+                    if (j > 0)
+                    {
+                        std::string_view prefix(firstArg.data() + start, j - start);
+                        parts.push_back({prefix, false});
```

Comment:
This is allocations again. We need to try to not allocate inside the algorithm as much as possible.